worker/src/main/java/gov/cms/ab2d/worker/processor/domainmodel/ContractEobManager.java

```diff
@@ -0,0 +1,231 @@
+package gov.cms.ab2d.worker.processor.domainmodel;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.parser.IParser;
+import gov.cms.ab2d.common.util.FHIRUtil;
+import gov.cms.ab2d.filter.FilterOutByDate;
+import gov.cms.ab2d.worker.adapter.bluebutton.ContractBeneficiaries;
+import gov.cms.ab2d.worker.processor.StreamHelper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.hl7.fhir.dstu3.model.ExplanationOfBenefit;
+import org.hl7.fhir.dstu3.model.OperationOutcome;
+import org.hl7.fhir.dstu3.model.Period;
+import org.hl7.fhir.dstu3.model.Resource;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.OffsetDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class ContractEobManager {
+    public enum ResourceStatus {
+        INVALID,
+        VALID,
+        UNKNOWN
+    }
+    static final List<Integer> ALL_MONTHS = List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12);
+
+    private final FhirContext fhirContext;
+
+    private final OffsetDateTime attTime;
+
+    private final Map<String, EobSearchResponse> unknownEobs = Collections.synchronizedMap(new HashMap<>());
+    private final Map<String, EobSearchResponse> validEobs = Collections.synchronizedMap(new HashMap<>());
+
+    private final boolean skipBillablePeriodCheck;
+    private final Date earliestDate;
+
+    public ContractEobManager(FhirContext fhirContext, boolean skipBillablePeriodCheck, Date earliestDate, OffsetDateTime attTime) {
+        this.fhirContext = fhirContext;
+        this.skipBillablePeriodCheck = skipBillablePeriodCheck;
+        this.earliestDate = earliestDate;
+        this.attTime = attTime;
+    }
+
+    public void writeValidEobs(StreamHelper helper) throws IOException {
+        IParser jsonParser = fhirContext.newJsonParser();
+        String payload = "";
+        int count = 0;
+        try {
+            for (Map.Entry<String, EobSearchResponse> response : validEobs.entrySet()) {
+                Iterator<Resource> resources = response.getValue().getResources().iterator();
+                while (resources.hasNext()) {
+                    Resource r = resources.next();
```

Comment:
Iterator.forEachRemaining won't work because you are removing items from the list during the loop as you are writing to the file. I extract the file writing to another method.